### DISCUSSION TITLE
Implement into immich mobile app

### DISCUSSION THREAD

**Discussion by mathwizmf2009** [simulot/immich-go#70] [General]
Is there any chance you could work with the actual immich project to implement some of your features into their app?

One specific idea is regarding your duplicate detection. I really like how your tool checks if the image should even be uploaded prior to actually uploading the file. This is not what the immich mobile (Android) app does. The app uploads everything again and has the server tell the app if it's a duplicate file after the fact. 

I made the mistake of rolling back my android app version, which caused my entire local phone library needing to be resynched. This is a painful process and unnecessary since everything is already on the server, but made worse by needing to actually re-upload all my files just for the app to determine that they're already on the server and can be skipped in the future.

**Comment by simulot** [comment]
Thank for you interest in my project.

> Is there any chance you could work with the actual immich project to implement some of your features into their app?

Alas, this isn't in my capabilities. Immich is using the Dart/Typescript/Node languages when I'm using the Go language. And more important development of mobile app is an other world. So, it won't happen soon.

On you specific issue, I have read that checking image signature on the mobile before uploading is on the roadmap. You can reach then immich development team on the discord server.
